src/testing/core/mocking/__tests__/mockLink.ts

```diff
@@ -1,7 +1,55 @@
 import gql from "graphql-tag";
-import { MockLink } from "../mockLink";
+import { MockLink, MockedResponse } from "../mockLink";
 import { execute } from "../../../../link/core/execute";
 
+/*
+These are Type check tests for ensuring MockedResponse.newData type works as expected
+*/
+// Untyped MockedResponse means newData is weakly typed
+const passingTesForNewDataUntypedMockedResponse: MockedResponse = {
+  request: {
+    query: gql`query A { a }`,
+  },
+
+  newData: ({ fake: { faker }}) => ({
+    data: {
+      pretend: faker,
+    },
+  }),
+}
+
+// @ts-expect-error 'data' still has to be an object
+passingTesForNewDataUntypedMockedResponse.newData = () => ({
+  data: "123",
+});
+
+// Strongly typed MockedResponse is much more developer friendly
```

Comment:
This is technically true of any type, so I'm not sure this comment is super helpful. I like how your other cases describe the intended behavior, so how about something like:

```suggestion
// Requires output type to match query type
```

Besides, I think this would make a better `test(...)` description anyways. See my other comment on converting these to use the `test`/`it` syntax.

Reply:
done